client/modules/datafiles/src/DatafilesModal/RenameModal/RenameModal.tsx

```diff
@@ -0,0 +1,190 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { Button, Modal, Form, Input } from 'antd';
+import {
+  useAuthenticatedUser,
+  useSelectedFiles,
+  useRename,
+} from '@client/hooks';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { TModalChildren } from '../DatafilesModal';
+
+export const RenameModalBody: React.FC<{
+  isOpen: boolean;
+  api: string;
+  system: string;
+  path: string;
+  handleCancel: () => void;
+}> = ({ isOpen, api, system, path, handleCancel }) => {
+  const queryClient = useQueryClient();
+  const [form] = Form.useForm();
+  const { selectedFiles } = useSelectedFiles(api, system, path);
+  const selectedFilesName = selectedFiles.length
+    ? selectedFiles
+    : [{ name: '' }];
+  const { user } = useAuthenticatedUser();
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  const defaultDestParams = useMemo(
+    () => ({
+      destApi: 'tapis',
+      destSystem: 'designsafe.storage.default',
+      destPath: encodeURIComponent('/' + user?.username),
+    }),
+    [user]
+  );
+
+  const [dest, setDest] = useState(defaultDestParams);
+  const { destApi, destSystem } = dest;
+  useEffect(() => setDest(defaultDestParams), [defaultDestParams]);
+
+  const { mutate } = useRename();
+  const updateFileName = useCallback(
+    (newName: string, path: string) => {
+      mutate({
+        src: { api, system, path, name: selectedFiles[0].name },
+        dest: { api: destApi, system: destSystem, path: path, name: newName },
+      });
+      handleClose();
+    },
+    [selectedFiles, mutate, destApi, destSystem, api, system]
+  );
+
+  const handleRenameClick = async () => {
+    const path: string = user?.username ?? '';
+    const newName = form.getFieldValue('newName');
+    updateFileName(newName, path);
+  };
+
+  const handleClose = useCallback(() => {
```

Comment:
We can probably remove this `handleClose` callback since unlike previews, there isn't any cached data that we need to get rid of when the modal leaves the screen.